internal/olm/client/client.go

```diff
@@ -223,22 +224,27 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 			return false, nil
 		}
 	}
-
 	err := wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
 	if err != nil && errors.Is(err, context.DeadlineExceeded) {
-		if depCheckErr := c.printDeploymentErrors(ctx, key, csv); depCheckErr != nil {
-			return fmt.Errorf("error printing operator resource errors: %v %v", err, depCheckErr)
+		result, err := c.checkDeploymentErrors(ctx, key, csv)
+		if len(result.Outputs) > 0 {
+			if err := result.PrintWithFormat("json-alpha1"); err != nil {
+				return err
+			}
+		}
+		if err != nil {
+			return err
 		}
 	}
 	return err
 }
 
-// TODO(btenneti) Refactor function to collect errors into customized error and return.
-// printDeploymentErrors function loops through deployment specs of a given CSV, and prints reason
+// checkDeploymentErrors function loops through deployment specs of a given CSV, and returns reason
 // in case of failures, based on deployment condition.
-func (c Client) printDeploymentErrors(ctx context.Context, key types.NamespacedName, csv olmapiv1alpha1.ClusterServiceVersion) error {
+func (c Client) checkDeploymentErrors(ctx context.Context, key types.NamespacedName, csv olmapiv1alpha1.ClusterServiceVersion) (resultutil.Result, error) {
```

Comment:
We shouldn't be using this `Result` type for anything but `bundle validate`. It semantically doesn't make sense (there is no "document" concept here), and is subject to change since its format is "alpha"; it might even be removed when we stabilize the `bundle validate -o json` format as one we can't use here.

Can you change this back to printing from a map? We can discuss defining a standard resource error reporting format for all client/query code as a follow-up.